tools/c7n_azure/c7n_azure/query.py

```diff
@@ -213,9 +214,21 @@ def source_type(self):
         return self.data.get('source', 'describe-azure')
 
     def resources(self, query=None):
-        key = self.get_cache_key(query)
-        resources = self.augment(self.source.get_resources(query))
-        self._cache.save(key, resources)
+        cache_key = self.get_cache_key(query)
+
+        resources = None
+        if self._cache.load():
```

Comment:
Cool yeah we needed to fix this, but I admit I am a little worried about what new issues we'll discover now, as none of us have ever used the caching in Azure.

Reply:
Changed default cache for Azure to memory because of the way Azure SDK works..

15 minute old cache can introduce regressions in the resources if they were change after cache was saved (via the portal e.g.)